formats/opentype.fathom

```diff
@@ -224,33 +302,228 @@ let byte_encoding_table = fun (platform : Repr platform_id) => {
+    /// Number of groupings which follow
+    num_groups <- u32be,
+    /// Array of SequentialMapGroup records.
+    groups <- array32 num_groups sequential_map_group,
+};
+
+/// Format 13: Many-to-one range mappings
+///
+/// This subtable provides for situations in which the same glyph is used for hundreds or even
+/// thousands of consecutive characters spanning across multiple ranges of the code space. This
+/// subtable format may be useful for “last resort” fonts, although these fonts may use other
+/// suitable subtable formats as well.
+///
+/// ## References
+///
+/// - [Microsoft's OpenType Spec: Format 13: Segment mapping to delta values](https://docs.microsoft.com/en-us/typography/opentype/spec/cmap#format-13-many-to-one-range-mappings)
+/// - [Apple's TrueType Reference Manual: `'cmap'` format 13](https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6cmap.html)
+let cmap_subtable_format13 = fun (platform : Repr platform_id) => {
+    /// Set to 0.
+    _reserved <- reserved u16be 0,
+    /// The length of the subtable in bytes (including the header)
+    length <- u32be,
+    /// The language ID of the subtable
+    language <- cmap_language_id32 platform,
+    /// Number of groupings which follow
+    num_groups <- u32be,
+    /// Array of ConstantMapGroup records.
+    groups <- array32 num_groups constant_map_group,
+};
+
+/// # Format 14: Unicode Variation Sequences
+///
+/// Subtable format 14 specifies the Unicode Variation Sequences (UVSes) supported by the font. A
+/// Variation Sequence, according to the Unicode Standard, comprises a base character followed by a
+/// variation selector. For example, <U+82A6, U+E0101>.
+///
+/// This subtable format must only be used under platform ID 0 and encoding ID 5.
+///
+/// ## References
+///
+/// - [Microsoft's OpenType Spec: Format 14: Segment mapping to delta values](https://docs.microsoft.com/en-us/typography/opentype/spec/cmap#format-14-unicode-variation-sequences)
+/// - [Apple's TrueType Reference Manual: `'cmap'` format 14](https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6cmap.html)
+let cmap_subtable_format14 = fun (platform : Repr platform_id) => fun (table_start : Pos) => {
+    /// The length of the subtable in bytes (including the header)
+    length <- u32be,
+    /// Number of variation Selector Records
+    num_var_selector_records <- u32be,
+    /// Array of VariationSelector records.
+    var_selector <- array32 num_var_selector_records (variation_selector table_start),
+};
 
 /// # Character Mapping subtable
 let character_map_subtable = fun (platform : Repr platform_id) => {
+    /// The start of the character mapping sub-table
+    table_start <- stream_pos,
     /// Format number of the subtable
     format <- u16be,
-    /// The length ot the subtable in bytes
-    length <- u16be,
-    // TODO: format-specific subtable data
     data <- match format {
         0 => byte_encoding_table platform,
```

Comment:
Could maybe rename this to `cmap_subtable_format0` for consistency 🤔

Reply:
Ahh good idea. I didn't follow you lead there with the naming because the names of the other tables were a bit unwieldy and I thought it would be more readable to refer to them by their format number.

Reply:
Yeah that makes sense - I though it would be nice to have some more, uh, ‘semantic’ names for them, but I hadn't yet got up to the more goofy names. Can definitely understand going with numbered names here. I'm also willing to revisit the names I've given other tables too, if you disagree.